File: frontend/src/api/logs.ts
import { invoke } from '@tauri-apps/api/core'
import type {
  RequestLogListResponse,
  RequestLogDetail,
  SystemLogListResponse,
  GatewaySettings,
  GatewaySettingsUpdate
} from '@/types/models'

export interface RequestLogQuery {
  page?: number
  page_size?: number
  cli_type?: string
  provider_name?: string
}

export interface SystemLogQuery {
  page?: number
  page_size?: number
  level?: string
  event_type?: string
  provider_name?: string
}

export const logsApi = {
  getSettings: async () => {
    const data = await invoke<{ debug_log: number }>('get_gateway_settings')
    return { data: { debug_log: !!data.debug_log } as GatewaySettings }
  },
  updateSettings: async (data: GatewaySettingsUpdate) => {
    await invoke('update_gateway_settings', { debugLog: data.debug_log })
    return { data: null }
  },

  listRequestLogs: async (params: RequestLogQuery) => {
    const data = await invoke<RequestLogListResponse>('get_request_logs', {
      page: params.page,
      pageSize: params.page_size,
      cliType: params.cli_type
    })
    return { data }
  },
  getRequestLog: async (id: number) => {
    const data = await invoke<RequestLogDetail>('get_request_log_detail', { id })
    return { data }
  },
  clearRequestLogs: async (before_timestamp?: number) => {
    await invoke('clear_request_logs')
    return { data: null }
  },

  listSystemLogs: async (params: SystemLogQuery) => {
    const data = await invoke<SystemLogListResponse>('get_system_logs', {
      page: params.page,
      pageSize: params.page_size,
      level: params.level,
      eventType: params.event_type,
      providerName: params.provider_name
    })
    return { data }
  },
  clearSystemLogs: async (before_timestamp?: number) => {
    await invoke('clear_system_logs')
    return { data: null }
  }
}

File: backend/app/api/v1/logs.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession
from typing import Optional

from app.core.database import get_db, get_log_db
from app.schemas.schemas import (
    RequestLogListResponse, RequestLogListItem, RequestLogDetail,
    SystemLogListResponse, SystemLogItem, ClearLogsRequest,
    GatewaySettingsResponse, GatewaySettingsUpdate
)
from app.services.log_service import LogService
from app.services.settings_service import SettingsService

router = APIRouter()


@router.get("/settings", response_model=GatewaySettingsResponse)
async def get_log_settings(db: AsyncSession = Depends(get_db)):
    service = SettingsService(db)
    return await service._get_gateway_settings()


@router.put("/settings")
async def update_log_settings(data: GatewaySettingsUpdate, db: AsyncSession = Depends(get_db)):
    service = SettingsService(db)
    await service.update_gateway_settings(data)
    return {"message": "Log settings updated"}


@router.get("/requests", response_model=RequestLogListResponse)
async def list_request_logs(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    cli_type: Optional[str] = None,
    provider_name: Optional[str] = None,
    success: Optional[bool] = None,
    db: AsyncSession = Depends(get_db),
    log_db: AsyncSession = Depends(get_log_db)
):
    service = LogService(db, log_db)
    logs, total = await service.list_request_logs(page, page_size, cli_type, provider_name, success)
    items = [
        RequestLogListItem(
            id=log.id,
            created_at=log.created_at,
            cli_type=log.cli_type,
            provider_name=log.provider_name,
            model_id=log.model_id,
            success=bool(log.success),
            status_code=log.status_code,
            elapsed_ms=log.elapsed_ms,
            input_tokens=log.input_tokens,
            output_tokens=log.output_tokens,
            client_method=log.client_method,
            client_path=log.client_path
        ) for log in logs
    ]
    return RequestLogListResponse(items=items, total=total, page=page, page_size=page_size)


@router.get("/requests/{log_id}", response_model=RequestLogDetail)
async def get_request_log(
    log_id: int,
    db: AsyncSession = Depends(get_db),
    log_db: AsyncSession = Depends(get_log_db)
):
    service = LogService(db, log_db)
    log = await service.get_request_log(log_id)
    if not log:
        from fastapi import HTTPException
        raise HTTPException(status_code=404, detail="Log not found")
    return RequestLogDetail(
        id=log.id,
        created_at=log.created_at,
        cli_type=log.cli_type,
        provider_name=log.provider_name,
        model_id=log.model_id,
        success=bool(log.success),
        status_code=log.status_code,
        elapsed_ms=log.elapsed_ms,
        input_tokens=log.input_tokens,
        output_tokens=log.output_tokens,
        client_method=log.client_method,
        client_path=log.client_path,
        client_headers=log.client_headers,
        client_body=log.client_body,
        forward_url=log.forward_url,
        forward_headers=log.forward_headers,
        forward_body=log.forward_body,
        provider_status=log.provider_status,
        provider_headers=log.provider_headers,
        provider_body=log.provider_body,
        response_status=log.response_status,
        response_headers=log.response_headers,
        response_body=log.response_body,
        error_message=log.error_message
    )


@router.delete("/requests")
async def clear_request_logs(
    data: ClearLogsRequest,
    db: AsyncSession = Depends(get_db),
    log_db: AsyncSession = Depends(get_log_db)
):
    service = LogService(db, log_db)
    count = await service.clear_request_logs(data.before_timestamp)
    return {"message": f"Cleared {count} request logs"}


@router.get("/system", response_model=SystemLogListResponse)
async def list_system_logs(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    level: Optional[str] = None,
    event_type: Optional[str] = None,
    provider_name: Optional[str] = None,
    db: AsyncSession = Depends(get_db),
    log_db: AsyncSession = Depends(get_log_db)
):
    service = LogService(db, log_db)
    logs, total = await service.list_system_logs(page, page_size, level, event_type, provider_name)
    items = [
        SystemLogItem(
            id=log.id,
            created_at=log.created_at,
            level=log.level,
            event_type=log.event_type,
            provider_name=log.provider_name,
            message=log.message,
            details=log.details
        ) for log in logs
    ]
    return SystemLogListResponse(items=items, total=total, page=page, page_size=page_size)


@router.delete("/system")
async def clear_system_logs(
    data: ClearLogsRequest,
    db: AsyncSession = Depends(get_db),
    log_db: AsyncSession = Depends(get_log_db)
):
    service = LogService(db, log_db)
    count = await service.clear_system_logs(data.before_timestamp)
    return {"message": f"Cleared {count} system logs"}
